backend/src/main/java/com/eventus/backend/services/ParticipationService.java

```diff
@@ -0,0 +1,52 @@
+package com.eventus.backend.services;
+
+import com.eventus.backend.models.Participation;
+import com.eventus.backend.models.User;
+import com.eventus.backend.repositories.ParticipationRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataAccessException;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class ParticipationService {
+
+    private ParticipationRepository partRepository;
+
+    @Autowired
+    public ParticipationService(ParticipationRepository partRepository) {
+       this.partRepository = partRepository;
+    }
+
+    @Transactional
+    public void saveParticipation(Participation participation) throws DataAccessException {
+        partRepository.save(participation);
+    }
+
+    public Optional<Participation> findParticipationById(Long id) {
+        return partRepository.findById(id);
+    }
+
+    public void deleteParticipation(Long id) {
+        partRepository.deleteById(id);
+    }
+
+    public List<Participation> findParticipationByUserId(Long userId,Pageable p) {
+        return partRepository.findByUser_IdEquals(userId,p);
```

Comment:
Codacy found an smell code here. Maybe a custon query would help, or looking for a better query naming would be enough.
![image](https://user-images.githubusercontent.com/72513925/158830311-079d6881-ddfd-4f7a-9b51-b44bca7760ff.png)